### ISSUE TITLE
Some ideas that may be valuable (or not)

### ISSUE THREAD

**Issue by rudloffl** [snipsco/Snips-Jeedom-Plugin-Issues#2]
# Only ideas, no direct request:
* Make snips less linear in his response, offer more than 1 possible response (e.g. Turn on the light --> "it's done" OR "OK, no problem"  OR "I'm on it !"...)
* When I say turn off the light, could he ask in which room
* Get some feedback from snips such has "I didn't understand" or "The action you are asking is not set in jeedom"
* If an intent is not defined or linked in jeedom, It would be great to lock them (and have the feed-back accordingly)

Again, no request, only ideas. I can share others if you want.

**Comment by valentinfage** [comment]
Thanks for your valuable ideas!
1/ less linear response, let us think about it, the thing is that for some intents you will use {#} for dynamic TTS hence so that might complexify it but for basic TTS response we could let the user add multiple responses and have some kind of random selection yes. A big feature we need to add for TTS is to be able to stream the response from your Sonos or Gcast installation rather than on your local device.

2/ Hmm tricky. In fact, in Snips console, when you create an intent such as "TurnOnLight" you can create slots (in this case its "house_room") and you can select that this slot is mandatory and configure a response. Every time you will ask "Turn on the light" it would say "in which room?" based on what you want it to say. However by doing this you cannot just say "Turn on the light" to turn on the light everywhere otherwise you would have to specify "turn on the light in the whole house" do you understand what I mean?

3/ "I didn't understand" is also tricky since its actually tricky to make Snips understand that it actually didn't understand your query but will see what we can do for that

4/ I'm not sure I've understood your 4th idea

Again, keep them coming ;) we already have a list of feature to implements (available here on that repo) but always happy to have new ones.

**Comment by rudloffl** [comment]
About #4,
* I don't use the skill "thermostatset" because I only have a non connected thermostat
* If snips detects a request belonging to that skills, it's anyway a dead end with no outcome (nothing will be achieved because nothing is set in Jeedom).
* Block that skill (or mark that skill as not used) on the jeedom side can be used has an information by snips to know that he did not understand what I wanted. Jeedom must feed back to snips what's going on.

**Comment by valentinfage** [comment]
Indeed, nice feature but we could maybe make it even simplier by basically separating intents into differents apps. 

In other words, instead of having one "JeedomOfficial" with 16 intents.
You would maybe have a certain number of AppJeedom such as "ThermostatJeedom" "LightJeedom" and so on that you choose to load (or not) onto your assistant to make the perfect custom made assistant for your usage. 
The only drawback with doing this is that the more we modify the JeedomOfficial assistant or break it into different Apps, the more you will have to change your configuration files but that should give more flexibility.

**Comment by rudloffl** [comment]
5/ if an intent is detected, and no slots set is matching the action set defined in Jeedom, Snips misunderstood the request

**Comment by valentinfage** [comment]
5/ hmmm but Snips and Jeedom are not related. Snips understand queries no matter what you configured in Jeedom. In other words if you try directly your voice query on the console.snips.ai using JeedomOfficial you can “speak” your query and see what it understood even though it is not connected to Jeedom. 
If fact if there’s a word that Snips doesn’t recognize it is an assistant training issue, it means that when the assistant has been trained (machine learning) no examples using “table” was used, however the closest thing was “tablette”. I can just add some sentences in the training exemples to re-train the assistant.

6/ In that case, “Allume X dans le salon” will be recognized as “TurnOnJeedom” intent, then the condition here to check will be house_room == “Salon”. 
I'll try to explain. 
"Allume la musique dans le salon" or "Allume squeezbox" dans le salon" is the same intent with only one slot value = "salon". In other words if you say one or the other the if condition will only apply for "Salon" which is the house-room slot but if you say squeezbox or music its the same thing.

**Comment by valentinfage** [comment]
@rudloffl 

We've made the JeedomOfficial App forkable which mean you can now edit it and remove any intents you don't want and/or modify training sentences and data slots. This should solve the issue for 4/ and 6/

Just read the Snips general documentation about how to create a good and powerful assistant beforehand :)

**Comment by rudloffl** [comment]
It would be great to have the possibility to chain requests (I think it's annoying to ask to turn on the light in 3 different rooms, and must say 3 times "Hey Snips")

The solution could be to be polite with Snips ! Let me explain:
* "Hey Snips"
** Ding  --> I got his attention
* "Turn on the light in room 1"
** ding action completed, waiting for next order or timeout
* "Turn on the light in room 2"
** ding action completed, waiting for next order or timeout
* "Turn on the light in room 3"
** ding action completed, waiting for next order or timeout
* "Thank you snips" --> I'm done

**Comment by valentinfage** [comment]
Hello @rudloffl

That's already possible with the dialogue session https://snips.gitbook.io/documentation/ressources/multi-turn-dialog

Will try to make a tutorial about it but the short answer is, yes you can already do it

**Comment by fwehrle** [comment]
Hi,
Instead of split intents in separate Apps, why don't you just fork the offical one, fork intents, and customize it like you want?
I deleted thermostat Intents, customize training sentences, and mostly customize house_room (to correspond with jeedom object/room) and device_name slots. Now, Jeedom and snips perfectly speak the same language.
